pkg/kubectl/apply/element.go

```diff
@@ -325,6 +326,40 @@ func IsAdd(e Element) bool {
 	return e.HasLocal() && !e.HasRemote()
 }
 
+// IsConflict returns true if the field represented by e should both have the recorded value
+// and remote value, but the values are not same
+func IsConflict(e Element) bool {
+	if e.HasRecorded() && e.HasRemote() {
+		var record map[string]interface{}
+		var remote map[string]interface{}
+		switch e.(type) {
+		case ListElement:
+			return !reflect.DeepEqual(e.(ListElement).ListElementData.GetRecordedList(), e.(ListElement).ListElementData.GetLocalList())
+		case MapElement:
+			record = e.(MapElement).MapElementData.GetRecordedMap()
+			remote = e.(MapElement).MapElementData.GetRemoteMap()
+		case TypeElement:
+			record = e.(TypeElement).MapElementData.GetRecordedMap()
+			remote = e.(TypeElement).MapElementData.GetRemoteMap()
+		case PrimitiveElement:
+			return !reflect.DeepEqual(e.(PrimitiveElement).GetRecorded(), e.(PrimitiveElement).GetRemote())
+		default:
+			record = e.(TypeElement).MapElementData.GetRecordedMap()
+			remote = e.(TypeElement).MapElementData.GetRemoteMap()
+		}
+
+		for k, v := range record {
+			if value, ok := remote[k]; ok {
+				if !reflect.DeepEqual(v, value) {
```

Comment:
I don't think this will work, the maps may have different keys.  We only want to check if they have the same keys with different values (recursively applied)

Reply:
The algorithm is native traversing the two maps. As we only look for the same key, it does not matter which map has more keys nor which map is outside loop. Starting traversing from the record as example,  if the key is exists in remote, which means the "two maps have same keys". Then  go DeepEqual for the values (DeepEqual works for this valuation: For same type, then valuate the value, otherwise return false)

Test example: 
```
	record := make(map[string] interface{})
	remote := make(map[string] interface{})
	record["A"] = "TEXT_A"
	record["B"] = "TEXT_B"
	record["C"] = "TEXT_C"
	record["M"] = "TestMess"
	record["X"] = 64

	remote["A"] = "REMO_A"
	remote["M"] = "TestMess"
	remote["X"] = struct {
			name  string 
			key string
		      }{name: "Name", key: "Key"}

	for k, v := range record {
 		if value, ok := remote[k]; ok {
			if !reflect.DeepEqual(v, value) {
				fmt.Printf("\nFound: Key: %v, Record: %v, Remote: %v", k, v, value)
			}
		}
	}
```
Output:
```
Found: Key: A, Record: TEXT_A, Remote: REMO_A
Found: Key: X, Record: 64, Remote: {Name Key}
```